composer/loggers/wandb_logger.py

```diff
@@ -158,7 +158,19 @@ def log_file_artifact(self, state: State, log_level: LogLevel, artifact_name: st
                                f"The artifact with name '{artifact_name}' will be stored as '{new_artifact_name}'."))
 
             extension = new_artifact_name.split(".")[-1]
-            artifact = wandb.Artifact(name=new_artifact_name, type=extension)
+
+            metadata = {f"timestamp/{k}": v for (k, v) in state.timestamp.state_dict().items()}
+            # if evaluating, also log the evaluation timestamp
+            if state.dataloader is not state.train_dataloader:
+                # TODO If not actively training, then it is impossible to tell from the state whether
+                # the trainer is evaluating or predicting. Assuming evaluation in this case.
+                metadata = {f"eval_timestamp/{k}": v for (k, v) in state.eval_timestamp.state_dict().items()}
```

Comment:
Comment is a bit confusing -- should it be logging both `timestamp/` and `eval_timestamp/` (which comment implies with "also") or just one of them? 

Also, if it's the latter, I'm personally in favor of doing something like
```
metadata = None
if state.dataloader is state.train_dataloader:
  metadata = ...
else:
  metadata = ...
```
so it only constructs the dict if necessary, but more minor nit and I'm not sure what's more consistent with rest of the code base

Reply:
Ah good catch -- I meant to update the metadata, rather than override it. Thanks!

My thoughts are that if you are evaluating, then you would want both the training and evaluation timestamp. But if you're just training, then skip the evaluation timestamp.